compiler/testData/ir/irText/firProblems/recursiveCapturedTypeInPropertyReference.txt

```diff
@@ -0,0 +1,68 @@
+      overridden:
+        public open fun hashCode (): kotlin.Int declared in kotlin.Any
+      $this: VALUE_PARAMETER name:<this> type:kotlin.Any
+    FUN FAKE_OVERRIDE name:toString visibility:public modality:OPEN <> ($this:kotlin.Any) returnType:kotlin.String [fake_override]
+      overridden:
+        public open fun toString (): kotlin.String declared in kotlin.Any
+      $this: VALUE_PARAMETER name:<this> type:kotlin.Any
+  CLASS CLASS name:AbstractSymbol modality:ABSTRACT visibility:public superTypes:[kotlin.Any]
+    $this: VALUE_PARAMETER INSTANCE_RECEIVER name:<this> type:<root>.AbstractSymbol<E of <root>.AbstractSymbol>
+    TYPE_PARAMETER name:E index:0 variance: superTypes:[<root>.Recursive<E of <root>.AbstractSymbol>; <root>.Something]
+    CONSTRUCTOR visibility:public <> () returnType:<root>.AbstractSymbol<E of <root>.AbstractSymbol> [primary]
+      BLOCK_BODY
+        DELEGATING_CONSTRUCTOR_CALL 'public constructor <init> () [primary] declared in kotlin.Any'
+        INSTANCE_INITIALIZER_CALL classDescriptor='CLASS CLASS name:AbstractSymbol modality:ABSTRACT visibility:public superTypes:[kotlin.Any]'
+    FUN name:foo visibility:public modality:FINAL <> ($this:<root>.AbstractSymbol<E of <root>.AbstractSymbol>, list:kotlin.collections.List<kotlin.Any>) returnType:kotlin.Unit
+      $this: VALUE_PARAMETER name:<this> type:<root>.AbstractSymbol<E of <root>.AbstractSymbol>
+      VALUE_PARAMETER name:list index:0 type:kotlin.collections.List<kotlin.Any>
+      BLOCK_BODY
+        VAR name:result type:kotlin.collections.List<<root>.AbstractSymbol<out <root>.Recursive<*>>> [val]
+          CALL 'public final fun map <T, R> (transform: kotlin.Function1<T of kotlin.collections.map, R of kotlin.collections.map>): kotlin.collections.List<R of kotlin.collections.map> [inline] declared in kotlin.collections' type=kotlin.collections.List<<root>.AbstractSymbol<out <root>.Recursive<*>>> origin=null
+            <T>: <root>.Recursive<*>
+            <R>: <root>.AbstractSymbol<out <root>.Recursive<*>>
+            $receiver: CALL 'public final fun filterIsInstance <R> (): kotlin.collections.List<@[NoInfer] R of kotlin.collections.filterIsInstance> [inline] declared in kotlin.collections' type=kotlin.collections.List<@[NoInfer] <root>.Recursive<*>> origin=null
+              <R>: <root>.Recursive<*>
+              $receiver: GET_VAR 'list: kotlin.collections.List<kotlin.Any> declared in <root>.AbstractSymbol.foo' type=kotlin.collections.List<kotlin.Any> origin=null
+            transform: PROPERTY_REFERENCE 'public abstract symbol: <root>.AbstractSymbol<R of <root>.Recursive> [val]' field=null getter='public abstract fun <get-symbol> (): <root>.AbstractSymbol<R of <root>.Recursive> declared in <root>.Recursive' setter=null type=kotlin.reflect.KProperty1<<root>.Recursive<*>, <root>.AbstractSymbol<out <root>.Recursive<*>>> origin=null
```

Comment:
Note that the only differences I can see now are:
* lack of `@NoInfer` attribute, which is out of scope of this change
* lack of `out` variance: `AbstractSymbol<out <root>.Recursive<*>>>` v.s. `AbstractSymbol<<root>.Recursive<*>>`
See the 2nd type argument of property reference where the original stack overflow happened.